app/src/main/java/org/schabi/newpipe/player/seekbarpreview/SeekbarPreviewThumbnailHolder.java

```diff
@@ -63,13 +59,12 @@ public synchronized void resetFrom(
         executorService.shutdown();
     }
 
-    private void resetFromAsync(
-            final int seekbarPreviewType,
-            final List<Frameset> framesets,
-            final UUID updateRequestIdentifier) {
-
+    private void resetFromAsync(final int seekbarPreviewType, final List<Frameset> framesets,
+                                final UUID updateRequestIdentifier) {
         Log.d(TAG, "Clearing seekbarPreviewData");
-        seekbarPreviewData.clear();
+        synchronized (seekbarPreviewData) {
```

Comment:
Why is `synchronized` now needed?

Reply:
Android's sparse arrays are not thread-safe.